api/service/staking/service.go

```diff
@@ -53,25 +65,46 @@ func (s *Service) Run() {
 }
 
 // DoService does staking.
-func (s *Service) DoService(peer *p2p.Peer) {
+func (s *Service) DoService(peer p2p.Peer) {
 	utils.GetLogInstance().Info("Staking with Peer")
 
 	// TODO(minhdoan): How to use the p2p or pubsub to send Staking Message to beacon chain.
 	// See below of how to create a staking message.
 }
 
-func (s *Service) createStakingMessage() *message.Message {
-	// TODO(minhdoan): Add signature and public key.
-	// To add public we need to assign a fake account to new node initially.
-	return &message.Message{
-		Type: message.MessageType_NEWNODE_BEACON_STAKING,
-		Request: &message.Message_Staking{
-			Staking: &message.StakingRequest{
-				Transaction: []byte{},
-				NodeId:      "",
+func (s *Service) getAccountState(beaconPeer p2p.Peer) *proto.FetchAccountStateResponse {
+	client := client.NewClient(beaconPeer.IP, beaconPeer.Port)
+	defer client.Close()
+	return client.GetBalance(crypto.PubkeyToAddress(s.accountKey.PublicKey))
+}
+
+func (s *Service) createStakingMessage(beaconPeer p2p.Peer) *message.Message {
+	accountState := s.getAccountState(beaconPeer)
+	toAddress := common.HexToAddress("0x4592d8f8d7b001e72cb26a73e4fa1806a51ac79d")
+	gasLimit := uint64(0)
+	gasPrice := big.NewInt(0)
+	tx := types.NewTransaction(
+		accountState.Nonce,
+		toAddress,
+		0, // beacon chain.
+		new(big.Int).SetBytes(accountState.Balance),
+		gasLimit,
```

Comment:
We should not stake the entire account balance :) It needs to take in a staking amount parameter.

Reply:
yeah, agree and i knew it. like i said the right design is to let miner user created signed tx in bytes and provide it to the new node for staking. it's a bit awkward to specify the amount of the staking, private key, public key though even the faucet contract does not allow us the capability to set how much we want to send free money/token to the testing/fake accounts.

That's why i created the issue https://github.com/harmony-one/harmony/issues/400

Let's not block things as we have a lot more work after PRs. You can remember how i wanted to approve your PR right away though i think it could need to be fixed.

Reply:
See here for the faucet contract https://github.com/harmony-one/harmony/blob/master/contracts/Faucet.sol

Reply:
Yes, but the staking contract allows for stake.
The PR needs signing library as we discussed?

Reply:
created yesterday https://github.com/harmony-one/harmony/issues/415

Reply:
And actually the library was against with the usage of constant fakeAccounts. it's not urgent now.